### PR TITLE
docs: update contributing and FAQ docs

### LINKED ISSUE #32: k8s API v1.21 and cdk8s-plus-21
This is just a question.

To use k8s API v1.19, v1.20 and cdk8s together,

Do I have to create my own `cdk8s-plus-19`, `cdk8s-plus-20`?

Currently, this repo provides only `cdk8s-plus-17`.

I was just wondered but still wondered even with [the documentation](https://cdk8s.io/docs/v1.0.0-beta.5/plus/).

Thank you.

### PR DESCRIPTION
**Pull request by Chriscbr**
Resolves https://github.com/cdk8s-team/cdk8s-plus-17/issues/32

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -126,9 +126,46 @@ Many times, this may result in some code duplication or redundancy. Even though
 
 > **Sometimes, a little duplication is better than a lot of dependencies.**
 
+## Making a pull request
+
+This repository is set up to maintain one cdk8s+ library for each of the **three
+latest versions** of kubernetes. Each library may have slightly different code.
+Because of that, contributing requires a few extra steps to make sure we can
+apply the feature or bug fix to all versions of the library.
```
Comment (iliapolo) [inline CONTRIBUTING.md]:
```suggestion
apply the feature or bug fix to all the appropriate libraries.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -126,9 +126,46 @@ Many times, this may result in some code duplication or redundancy. Even though
...
+
+We currently track each version on a separate branch:
+
+- k8s-22/main
+- k8s-21/main
+- k8s-20/main
+
+In most cases, all you need to do is make a change to the latest version of
+cdk8s+. To do this, you can just follow the standard process for making a pull
```
Comment (iliapolo) [inline CONTRIBUTING.md]:
```suggestion
In most cases, all you need to do is make a change to the library corresponding to latest version of
kubernetes (i.e to the `k8s-22/main` branch). To do this, you can just follow the standard process for making a pull
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -126,9 +126,46 @@ Many times, this may result in some code duplication or redundancy. Even though
...
+automatically updated.
+
+When you clone the repo, it will check out the branch corresponding to the
+latest version (k8s-22/main), and when you create the pull request on GitHub, it
+should automatically target this branch. We will take care of backporting your
+change to the other branches corresponding to cdk8s-plus-21 and cdk8s-plus-20
+once the changes are merged to cdk8s-plus-22.
+
+> Note: In some situations, you might only want to make a change to an older
+version of cdk8s+. For example, `IngressV1Beta` is not available in
```
Comment (iliapolo) [inline CONTRIBUTING.md]:
```suggestion
> Note: In some situations, you might only want to make a change to a library targeting an older kubernetes version. For example, `IngressV1Beta` is not available in
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -126,9 +126,46 @@ Many times, this may result in some code duplication or redundancy. Even though
...
+cdk8s-plus-22, so changing it would require making a change to cdk8s-plus-21 and
+cdk8s-plus-20. If you need to make a pull request to a version of cdk8s-plus
+that isn't the latest version, then **the branch name of your pull request must
+start with `k8s-XX/`** where XX is the version number. For example, to make a
+pull request to cdk8s-plus-21, then you could name the branch `k8s-21/bug-fix`.
```
Comment (iliapolo) [inline CONTRIBUTING.md]:
```suggestion
pull request to cdk8s-plus-21, you could name the branch `k8s-21/bug-fix`.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -126,9 +126,46 @@ Many times, this may result in some code duplication or redundancy. Even though
...
+When you submit the pull request on GitHub, make sure the target branch matches
+your branch name (in this example, it would be `k8s-21/main`).
+
```
Comment (iliapolo) [inline CONTRIBUTING.md:167]: Add a note that in such cases, the PR needs to target the latest branch that you fix applies for. 
In this example, only a PR to `k8s-21/main` is required, and we will backport it to `k8s-20/main`.

File: README.md [hunk 6]
```diff
@@ -3,21 +3,75 @@
...
+[![cdk8s-plus-20](https://img.shields.io/github/workflow/status/cdk8s-team/cdk8s-plus/release-k8s.20?label=cdk8s-plus-20&logo=GitHub)](https://github.com/cdk8s-team/cdk8s-plus/actions/workflows/release-k8s.20.yml)
 
-**cdk8s+** is a software development framework that provides high level abstractions for authoring Kubernetes applications.
-Built on top of the auto generated building blocks provided by [cdk8s](../cdk8s), this library includes a hand crafted *construct*
-for each native kubernetes object, exposing richer API's with reduced complexity.
+| k8s version | npm (JS/TS) | PyPI (Python) | NuGet (C#) | Maven (Java) | Go |
+| --- | --- | --- | --- | --- | --- |
+| 1.20.0 | [Link](https://www.npmjs.com/package/cdk8s-plus-20) | [Link](https://pypi.org/project/cdk8s-plus-20/) | [Link](https://www.nuget.org/packages/Org.Cdk8s.Plus20) | [Link](https://search.maven.org/artifact/org.cdk8s/cdk8s-plus-20) | [Link](https://pkg.go.dev/github.com/cdk8s-team/cdk8s-plus-go/cdk8splus20) |
```
Comment (iliapolo) [inline README.md]: Maybe make the Go links point to the relevant GitHub tag? This way all links point to the published artifact.

File: README.md [hunk 7]
```diff
@@ -3,21 +3,75 @@
...
 
-## License
+## :raised_hand: Contributing
+
+If you'd like to add a new feature or fix a bug, please visit
+[CONTRIBUTING.md](CONTRIBUTING.md)!
+
+## :question: FAQ
+
+### What's the difference between cdk8s+ 20, cdk8s+ 21, and cdk8s+ 22?
```
Comment (iliapolo) [inline README.md]: Let's use the accurate package name when explaining technical stuff. 

```suggestion
### What's the difference between cdk8s-plus-20, cdk8s-plus-21, and cdk8s-plus-22?
```

File: README.md [hunk 8]
```diff
@@ -3,21 +3,75 @@
...
+
+These are separately vended libraries that all provide the same Kubernetes
+abstractions (as much as possible). The essential difference is that cdk8s+ 20
+generates Kubernetes YAMLs for the Kubernetes 1.20.0 API specification, cd8s+ 21
+generates Kubernetes YAMLs for the Kubernetes 1.21.0 API specification, and
+cdk8s+ 22 generates Kubernetes YAMLs for the Kubernetes 1.22.0 API
+specification.
```
Comment (iliapolo) [inline README.md]:
```suggestion
These are separately vended libraries that each target a different kubernetes version, marked by the `-XX` suffix.
For example, `cdk8s-plus-22` targets kubernetes version `1.22.0`. 
```

File: README.md [hunk 9]
```diff
@@ -3,21 +3,75 @@
...
+
+We offer the three latest versions because each Kubernetes version can
+potentially add, remove, or deprecate entire APIs. In addition, the Kubernetes
+project guarantees that they will maintain release branches of the three latest
+versions (currently 1.22, 1.21, 1.20).
```
Comment (iliapolo) [inline README.md]:
```suggestion
We offer a dedicated version per Kubernetes version to allow users to match their manifests to the Kubernetes version they are operating. This way, users are only exposed to a set of capabilities offered by their specific cluster, preventing deployment errors caused by version mismatches. 

According to the Kubernetes [SLA](reference needed), every unstable resource will be available for at least 3 versions before being removed, so we will always offer libraries for the the three latest versions. 
```

File: README.md [hunk 10]
```diff
@@ -3,21 +3,75 @@
...
+newly added in Kubernetes 1.X.
+
+The [Kubernetes Deprecation Policy] specifies that stable and beta APIs will
+continue being supported for 3 more versions after the version in which they are
+deprecated. Because of this, some versions of cdk8s+ may have multiple versions
+of resources available (such as `Ingress` and `IngressV1Beta`) since even though
+one is deprecated, it is still possible to deploy either to your cluster.
+
+[Kubernetes Deprecation Policy]:
+https://kubernetes.io/docs/reference/using-api/deprecation-policy/
```
Comment (iliapolo) [inline README.md]: I think we should split this up to two questions:

1. I operate Kubernetes version X - which cdk8s+ library should I be using?

This should also give instructions (with a disclaimer) on what to do if a `cdk8s-plus-X` doesn't exist.

2. I'm using cdk8s-plus-X - which kubernetes versions will my manifest work on?

This should be broken down to:

- Deploying to kubernetes version > X
- Deploying to kubernetes version < X

Comment (iliapolo) [reply README.md]: Lets see how this looks like with the new structure and we'll do a second pass after that.

**Review by iliapolo — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (Chriscbr) [reply README.md]: Done.

File: README.md (hunk 10 above)
Comment (Chriscbr) [reply README.md]: Removed this FAQ in favor of moving the FAQ to cdk8s.io

**Review by iliapolo — APPROVED** [review]

**Pull request merged**